### DISCUSSION TITLE
No Exception on empty append

### DISCUSSION THREAD

**Discussion by abbasudo** [laravel/framework#49496] [Ideas]
### Laravel Version

10.38.2

### PHP Version

8.1.4

### Description

laravel dont throw an exception when empty string was appended as below:
```php
$user = User::first()->append(['']);
```
this query would return an empty key with a null value:
```php
...
    "created_at": "2023-07-15T15:13:40.000000Z",
    "updated_at": "2023-07-15T15:13:40.000000Z",
    "": null
...
```

### Steps To Reproduce

 - create new laravel app
 - create new controller
 - add `return User::first()->append(['']);`
 - add the controller to routes

**Comment by sonja-turo** [comment]
As far as I am aware, an empty string is a perfectly valid array key from a PHP standpoint. It probably doesn't have a lot of useful use cases ( I can't think of any), but it's still valid. If the_framework_ has a valid reason for excluding empty strings as keys for appended attributes, then it should probably be documented somewhere.

For the most part though, it should be up to your application to decide if that's a valid key for your model.
